client/my-sites/plans-features-main/hooks/use-intent-from-site-meta.ts

```diff
@@ -0,0 +1,35 @@
+import { useSiteIntent } from '@automattic/data-stores';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { Intent } from 'calypso/my-sites/plans-features-main/hooks/use-plan-types-with-intent';
+
+interface IntentFromSiteMeta {
+	processing: boolean;
+	intent: Intent | null | undefined;
+}
```

Comment:
IMO, since this involves a network call, we should also expose `isError` so that the component can handle it gracefully.

Reply:
Same with https://github.com/Automattic/wp-calypso/pull/77667#discussion_r1239997834 When it comes to wanting to be explicit about `isError`, if we ever want to, then we'll add it. It feels like YAGNI right now. If it fails, it fails gracefully with a `null` intent, at least that's what's "intended" (gotta love "intent" word).